### ISSUE TITLE
segfault in numpy.subtract on MacOS X 10.5 (migrated from Trac #1895)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3443]
Original ticket http://projects.scipy.org/numpy/ticket/1895
Reported 2011-07-05 by atmention:r-owen, assigned to atmention:rgommers.

Using the attached pickled data file and script I'm reliably getting a segfault in numpy.subtract on MacOS X 10.5.8 with a fresh Mac Python 2.7.2 32-bit from python.org and a fresh install of the official numpy 1.6.0 Mac binary.

The self-test reports no errors (3 known fail, 2 skip, 3179 tests in all).

Other tid-bits that may be relevant:
- I started with a larger data file and kept trying different subregions. The attached data file is about the smallest subregion I could find that showed the problem. Below that the crash went away.
- If I convert it to float32 I still get the crash.
- I don't see the crash on MacOS X 10.6.2 using the same python and numpy binary installer.

**Comment by thouis** [comment]
Attachment in Trac by atmention:r-owen, 2011-07-05: [dataArr](http://projects.scipy.org/numpy/attachment/ticket/1895/dataArr)

**Comment by thouis** [comment]
Attachment in Trac by atmention:r-owen, 2011-07-05: [numpycrash.py](http://projects.scipy.org/numpy/attachment/ticket/1895/numpycrash.py)

**Comment by thouis** [comment]
Attachment in Trac by atmention:r-owen, 2011-07-05: [crash report.txt](http://projects.scipy.org/numpy/attachment/ticket/1895/crash report.txt)

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-07-11

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-07-11

Russell, can you confirm that this is fixed too in 1.6.1rc3?  (Just tagged, not uploaded yet)

**Comment by thouis** [comment]
Comment in Trac by atmention:r-owen, 2011-07-11

The problem appears to be fixed in 1.6.1rc3. I first ran the test code using numpy 1.6.0 on MacOS X 10.8.5 and got a seg fault. I then then installed numpy 1.6.1rc3 and the test code ran fine (no seg fault). The self-tests also all passed.

I tried this ticket's test code and ticket #1896's test code on numpy 1.6.1rc3 on MacOS X 10.6.8 and they ran fine, plus the numpy self-tests all passed. Thanks for the speedy fix!
